### PR TITLE
Fix mitter length cutoff and near plane intersection

### PR DESCRIPTION
**Pull request by IOTRFAFH**

### CODE REVIEW

File: worldwind/src/commonMain/kotlin/earth/worldwind/render/program/TriangleShaderProgram.kt [hunk 1]
```diff
@@ -27,41 +28,72 @@ open class TriangleShaderProgram : AbstractShaderProgram() {
...
                     vec2 tangent = normalize(AB + BC);
+                    vec2 point = normalize(AB - BC);
                     
                     vec2 miter = vec2(-tangent.y, tangent.x);
                     vec2 normalA = vec2(-AB.y, AB.x);
                     float miterLength = 1.0 / max(dot(miter, normalA), invMiterLengthCutoff);
                     
-                    gl_Position = pointBScreen;
-                    gl_Position.xy = gl_Position.xy + (corner * miter * lineWidth * miterLength) / screen.xy;
+                    float cornerX = floor((pointB.w + 0.5) * 0.5);
```
Comment (sopyer) [inline worldwind/src/commonMain/kotlin/earth/worldwind/render/program/TriangleShaderProgram.kt:86]: PointB.w*0.5 + 0.25?

File: worldwind/src/commonMain/kotlin/earth/worldwind/render/program/TriangleShaderProgram.kt [hunk 2]
```diff
@@ -27,41 +28,72 @@ open class TriangleShaderProgram : AbstractShaderProgram() {
...
+                    float cornerY = mod(pointB.w, 2.0);
```
Comment (sopyer) [inline worldwind/src/commonMain/kotlin/earth/worldwind/render/program/TriangleShaderProgram.kt:87]: ((pointB.w + 0.5) * 0.5) - cornerX?

File: worldwind/src/commonMain/kotlin/earth/worldwind/render/program/TriangleShaderProgram.kt [hunk 3]
```diff
@@ -27,41 +28,72 @@ open class TriangleShaderProgram : AbstractShaderProgram() {
...
+                    cornerX = 2.0 * cornerX - 1.0;
+                    cornerY = 2.0 * cornerY - 1.0;
+                    if (abs(miterLength - (1.0 / invMiterLengthCutoff)) < eps && cornerY * dot(miter, point) > 0.0) {
+                      // trim the corner
+                        gl_Position.xy = pointBScreen.w * (pointBScreen.xy - (cornerX * cornerY * lineWidth * normalA) / screen.xy);
```
Comment (sopyer) [inline worldwind/src/commonMain/kotlin/earth/worldwind/render/program/TriangleShaderProgram.kt:92]: screen.xy - dcalers, screen.zw - inverse

Comment (IOTRFAFH) [reply worldwind/src/commonMain/kotlin/earth/worldwind/render/program/TriangleShaderProgram.kt]: done

File: worldwind/src/commonMain/kotlin/earth/worldwind/render/program/TriangleShaderProgram.kt (hunk 2 above)
Comment (IOTRFAFH) [reply worldwind/src/commonMain/kotlin/earth/worldwind/render/program/TriangleShaderProgram.kt]: done

**Pull request merged**